stl/inc/xutility

```diff
@@ -3634,11 +3640,19 @@ namespace ranges {
                                       { _STD get<0>(__t) } -> convertible_to<const tuple_element_t<0, _Ty>&>;
                                       { _STD get<1>(__t) } -> convertible_to<const tuple_element_t<1, _Ty>&>;
                                   };
+#endif // !_HAS_CXX23
 
+    // clang-format off
     template <class _Ty, class _First, class _Second>
-    concept _Pair_like_convertible_from = (!range<_Ty>) && _Pair_like<_Ty> && constructible_from<_Ty, _First, _Second>
-                                       && _Convertible_to_non_slicing<_First, tuple_element_t<0, _Ty>>
-                                       && convertible_to<_Second, tuple_element_t<1, _Ty>>;
+    concept _Pair_like_convertible_from =
+        !range<_Ty>
+#if _HAS_CXX23
+        && !is_reference_v<_Ty>
+#endif // _HAS_CXX23
+        && _Pair_like<_Ty> && constructible_from<_Ty, _First, _Second>
+        && _Convertible_to_non_slicing<_First, tuple_element_t<0, _Ty>>
+        && convertible_to<_Second, tuple_element_t<1, _Ty>>;
+    // clang-format on

```

Comment:
We can preserve #3483's enabling of clang-format here by preserving the parens around `!range<_Ty>` and adding parens to `!is_reference_v<_Ty>`.